inference-gateway/src/events_reporter.rs

```diff
@@ -0,0 +1,146 @@
+use anyhow::{Context, Result};
+use chrono::{DateTime, Utc};
+use log::info;
+use pgmq::PGMQueueExt;
+use serde::{Deserialize, Serialize};
+use sqlx::{Pool, Postgres};
+use std::{env, time::Duration};
+use tokio::time::interval;
+use uuid::Uuid;
+
+use crate::db::connect;
+use crate::errors::DatabaseError;
+
+pub fn split_events(events: Vec<Events>, max_size: usize) -> Vec<Message> {
+    events
+        .chunks(max_size)
+        .map(|chunk| Message {
+            id: Uuid::new_v4().to_string(),
+            message: chunk.to_vec(),
+        })
+        .collect()
+}
+
+pub async fn get_usage(
+    dbclient: &Pool<Postgres>,
+    start_time: DateTime<Utc>,
+    end_time: DateTime<Utc>,
+) -> Result<Vec<Events>, DatabaseError> {
+    let rows = sqlx::query_as!(
+        UsageData,
+        r#"
+        SELECT
+          organization_id,
+          instance_id,
+          duration_ms,
+          prompt_tokens,
+          completion_tokens,
+          model,
+          completed_at
+        FROM inference.requests
+        WHERE
+          completed_at >= $1 AND completed_at < $2
+        "#,
+        start_time,
+        end_time
+    )
+    .fetch_all(dbclient)
+    .await?;
+
+    Ok(rows_to_events(rows))
+}
+
+pub fn rows_to_events(rows: Vec<UsageData>) -> Vec<Events> {
+    rows.into_iter()
+        .map(|row| Events {
+            idempotency_key: Uuid::new_v4().to_string(),
```

Comment:
this should be deterministic from the time range being reported on. for example if reporting for 1 hour of data, it should represent which hour (in UTC) that was being reported on.